Model/Giftcard/Response/Giftcard.php

```diff
@@ -136,26 +139,117 @@ public function set(TransactionResponse $response, CartInterface $quote)
+            $order = $order ?? $this->createOrder();
+            $this->updateQuotePaymentAmounts();
         }
-        if (!empty($this->response->getSubCodeMessage())) {
-            return $this->response->getSubCodeMessage();
+
+        if ($order) {
+            $this->quote->setOrigOrderId($order->getEntityId());
         }
 
-        if (isset($this->response->getFirstError()['ErrorMessage'])) {
-            return $this->response->getFirstError()['ErrorMessage'];
+        $this->logger->addDebug(sprintf(
+            '[Giftcard] | [Response] | [%s:%s] - Create Order From Quote | QuoteId: %s - OrderId: %s',
+            __METHOD__,
+            __LINE__,
+            $this->quote->getId(),
+            $order->getEntityId()
+        ));
+
+        $this->quote->setIsActive(true);
+        $this->quote->save();
+
+        return $order;
+    }
+
+    /**
+     * Ensure customer email is set in the quote.
+     */
+    protected function ensureCustomerEmail(): void
+    {
+        if (!$this->quote->getCustomerEmail()) {
+            $this->quote->setCustomerEmail($this->quote->getBillingAddress()->getEmail());
         }
-        if (isset($this->response->data()['Status']['Code']['Description'])) {
-            return $this->response->data()['Status']['Code']['Description'];
+    }
+
+    /**
+     * Retrieve the existing order if it exists.
+     *
+     * @return OrderInterface|null
+     */
+    protected function getExistingOrder(): ?OrderInterface
+    {
+        $orderId = $this->quote->getOrigOrderId();
+        if (!$orderId) {
+            return null;
         }
-        return '';
+
+        try {
+            return $this->orderRepository->get($orderId);
+        } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
+            return null;
+        }
+    }
+
+    /**
+     * Create order from quote
+     *
+     * @return AbstractExtensibleModel|OrderInterface|object|null
+     * @throws LocalizedException
+     */
+    protected function createOrder()
+    {
+        $this->quote->collectTotals();
+        return $this->quoteManagement->submit($this->quote);
+    }
+
+    /**
+     * Update the quote's payment amounts based on the currency rate.
+     */
+    protected function updateQuotePaymentAmounts()
+    {
+        $rate = 1.0;
+        $store = $this->quote->getStore();
+        $currency = $store->getCurrentCurrencyCode();
+        if ($currency != $store->getBaseCurrencyCode()) {
+            $rate = $store->getBaseCurrency()->getRate($currency);
+        }
+
+        $amountPaid = $this->response->getAmount();
+        $baseAmountPaid = (float)$amountPaid / (float)$rate;
+
+        $this->quote->setBuckarooAlreadyPaid(
+            $this->quote->getBuckarooAlreadyPaid() + $amountPaid
+        );
+
+        $this->quote->setBaseBuckarooAlreadyPaid(
+            $this->quote->getBaseBuckarooAlreadyPaid() + $baseAmountPaid
+        );
```

Comment:
why do you save these values in the quote?

Reply:
Without setting the buckaroo already paid amount the pay remainder it is grand total

![image](https://github.com/buckaroo-it/Magento2/assets/106905746/fbb05ee9-452f-419f-8317-ecb2974906fe)

https://plaza.buckaroo.nl/Transaction/Transactions/Details?transactionKey=05F737EC9A0546EF9BBCB0C1F1C8C017

I will check the function that sets pay remainder amount on payment method request.

Reply:
these values should be taken from the group transaction table

Reply:
https://www.awesomescreenshot.com/video/24189489?key=526a3cf47be3b8c585147ceba72a6e74

The code was used to directly identify from the quote object whether the command is a group transaction or not in order to not increment the order ID at place order, because it was already incremented when the order was created at the first successful transaction with a gift card 

 Plugin/QuotePlugin.php.

I modified the logic to identify if an order is a group transaction based on the group_transactions table.